### PR TITLE
[🚀esl-share]: popup internal implementation follow-up

### PR DESCRIPTION
**Pull request by dshovchko**
Closes: #1784

### CODE REVIEW

File: src/modules/esl-share/core/esl-share-trigger.ts [hunk 1]
```diff
@@ -1,37 +1,157 @@
...
+import type {ESLSharePopupActionParams} from './esl-share-popup';
 
 /**
  * ESLShareTrigger component
  * @author Dmytro Shovchko
  *
- * ESLShareTrigger is an extension of {@link ESLTrigger} that
- * - controls an internal popup of {@link ESLShare} module in `popup` rendering mode.
- * - forwards the sharing attributes from the parent share {@link ESLShare} component to its associated instance of {@link ESLPopup}
+ * ESLShareTrigger is an component that allows triggering {@link ESLSharePopup} instance state changes.
```
Comment (yadamskaya) [inline src/modules/esl-share/core/esl-share-trigger.ts:15]:
```suggestion
 * ESLShareTrigger is a component that allows triggering {@link ESLSharePopup} instance state changes.
```

File: src/modules/esl-share/core/esl-share-trigger.ts [hunk 2]
```diff
@@ -1,37 +1,157 @@
...
+
+  /** Sets initial a11y attributes */
+  public initA11y(): void {
+    if (!this.hasAttribute('role')) this.setAttribute('role', 'button');
+    if (this.getAttribute('role') === 'button' && !this.hasAttribute('tabindex')) {
+      this.setAttribute('tabindex', '0');
+    }
+  }
+
+  /** Merge params to pass to the toggleable */
```
Comment (yadamskaya) [inline src/modules/esl-share/core/esl-share-trigger.ts:86]:
```suggestion
  /** Merges params to pass to the toggleable */
```

File: src/modules/esl-tooltip/core/esl-tooltip.shape.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import type {ESLPopupTagShape} from '../../esl-popup/core/esl-popup.shape';
+import type {ESLTooltip} from './esl-tooltip';
+
+/**
+ * Tag declaration interface of {@link ESLTooltip} element
+ * Used for TSX declaration
+ */
+export interface ESLTooltipTagShape<T extends ESLTooltip = ESLTooltip> extends ESLPopupTagShape<T> {
+  /** Disable arrow at Tooltip */
```
Comment (julia-murashko) [inline src/modules/esl-tooltip/core/esl-tooltip.shape.ts:9]:
```suggestion
  /** Disables arrow at Tooltip */
```

**Review by julia-murashko — APPROVED** [review]

File: src/modules/esl-share/core/esl-share-popup.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+   * Forwards share attributes from the activator (or its parents) to the {@link ESLSharePopup} instance.
+   * Skips empty attributes
+   */
+  protected forwardAttributes(): void {
+    const {activator} = this;
+    if (!activator) return;
+    ESLSharePopup.forwardedAttrs.forEach((name) => {
+      const el = activator.closest(`[${name}]`);
+      const value = el && el.getAttribute(name);
+      if (value) this.$$attr(name, value);
```
Comment (ala-n) [inline src/modules/esl-share/core/esl-share-popup.ts:81]:
```suggestion
      if (el) return;
      this.$$attr(name, el.getAttribute(name));
```

File: src/modules/esl-share/core/esl-share-trigger.ts [hunk 5]
```diff
@@ -1,37 +1,157 @@
...
  * ESLShareTrigger component
  * @author Dmytro Shovchko
  *
- * ESLShareTrigger is an extension of {@link ESLTrigger} that
- * - controls an internal popup of {@link ESLShare} module in `popup` rendering mode.
- * - forwards the sharing attributes from the parent share {@link ESLShare} component to its associated instance of {@link ESLPopup}
+ * ESLShareTrigger is an component that allows triggering {@link ESLSharePopup} instance state changes.
  */
-export class ESLShareTrigger extends ESLTrigger {
+export class ESLShareTrigger extends ESLBaseElement {
```
Comment (ala-n) [inline src/modules/esl-share/core/esl-share-trigger.ts:17]: Is it so different from the ESLTrigger?

**Review by fshovchko — APPROVED** [review]

File: src/modules/esl-share/core/esl-share-popup.ts [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+export interface ESLSharePopupActionParams extends TooltipActionParams {
+  /** list of social networks or groups of them to display */
+  list?: string;
+}
+
+/**
+ * ESLSharePopup component
+ * @author Dmytro Shovchko
+ *
+ * ESLSharePopup is an extension of {@link ESLPopup} that
```
Comment (fshovchko) [inline src/modules/esl-share/core/esl-share-popup.ts:22]:
```suggestion
 * ESLSharePopup is an extension of {@link ESLPopup} that:
```

File: src/modules/esl-share/core/esl-share-trigger.ts [hunk 7]
```diff
@@ -1,37 +1,157 @@
...
   public static override is = 'esl-share-trigger';
 
-  /** List of attributes to forward from the host to the target share {@link ESLPopup} */
-  public static forwardedAttrs = ['share-title', 'share-url'];
+  /**
+   * List of social networks or groups of them to display (all by default).
+   * The value - a string containing the names of the buttons or groups (specified with
+   * the prefix group:) separated by spaces.
```
Comment (fshovchko) [inline src/modules/esl-share/core/esl-share-trigger.ts:23]:
```suggestion
   * the prefix "group:") separated by spaces.
```

File: src/modules/esl-share/core/esl-share-trigger.ts (hunk 5 above)
Comment (dshovchko) [reply src/modules/esl-share/core/esl-share-trigger.ts:17]: Probably not, not much.
Is it worth a try? Or shall we discuss it first?

File: src/modules/esl-share/core/esl-share-popup.ts (hunk 4 above)
Comment (abarmina) [reply src/modules/esl-share/core/esl-share-popup.ts:81]: most likely you meant to write: 
```typescript
if (!el) return;
```

**Review by abarmina — APPROVED** [review]

File: src/modules/esl-tooltip/core/esl-tooltip.shape.ts (hunk 3 above)
Comment (dshovchko) [reply src/modules/esl-tooltip/core/esl-tooltip.shape.ts:9]: It seems to me that we use third-person singular verbs only for methods description, doesn't it?

File: src/modules/esl-share/core/esl-share-trigger.ts [hunk 8]
```diff
@@ -1,37 +1,103 @@
...
+    ESLSharePopup.show(actionParams);
+  }
 
-    this.forwardAttributes();
+  /** Hides popup with passed params */
+  public override hideTarget(params: ESLToggleableActionParams = {}): void {
+    const actionParams = this.mergeToggleableParams({}, params);
+    ESLSharePopup.hide(actionParams);
   }
+}
```
Comment (ala-n) [inline src/modules/esl-share/core/esl-share-trigger.ts:94]: Do we still need to override it, even when the `$target` is declared constantly to ESLSharePopup ?

File: src/modules/esl-share/core/esl-share-trigger.ts [hunk 9]
```diff
@@ -1,37 +1,103 @@
...
+  }
+
+  /** The base language of the element */
+  protected get currentLang(): string {
+    const el = this.closest('[lang]');
+    return (el) ? (el as HTMLElement).lang : '';
+  }
+
+  /** Container element that defines bounds of popups visibility */
+  protected get containerEl(): HTMLElement | undefined {
```
Comment (ala-n) [inline src/modules/esl-share/core/esl-share-trigger.ts:54]:
```suggestion
  protected get $containerEl(): HTMLElement | undefined {
```

File: src/modules/esl-share/core/esl-share-trigger.ts [hunk 10]
```diff
@@ -1,37 +1,103 @@
...
-  public static forwardedAttrs = ['share-title', 'share-url'];
+  /**
+   * List of social networks or groups of them to display (all by default).
+   * The value - a string containing the names of the buttons or groups (specified with
+   * the prefix `group:`) separated by spaces.
+   * @example "facebook reddit group:default"
+   * */
+  @attr({defaultValue: 'all'}) public list: string;
+  /** Popup state marker */
+  @boolAttr() public popupShown: boolean;
```
Comment (ala-n) [inline src/modules/esl-share/core/esl-share-trigger.ts:26]: TODO: delete, replace with correct `this.isTargetActive` processing

Comment (dshovchko) [reply src/modules/esl-share/core/esl-share-trigger.ts]: Removed as redundant

File: src/modules/esl-share/core/esl-share-trigger.ts (hunk 8 above)
Comment (dshovchko) [reply src/modules/esl-share/core/esl-share-trigger.ts:96]: I guess they are not needed after resolving the issue #1677

**Review by ala-n — APPROVED** [review]

**Review by fshovchko — APPROVED** [review]

**Review by HenadzV — APPROVED** [review]

**Review by abarmina — APPROVED** [review]

**Pull request merged**